rust/src/transformations/make_stable_expr/namespace_str/expr_strptime/mod.rs

```diff
@@ -0,0 +1,115 @@
+/// * `input_domain` - Expr domain
+/// * `input_metric` - The metric under which neighboring LazyFrames are compared
+/// * `expr` - The str.strptime expression
+pub fn make_expr_strptime<M: OuterMetric>(
+    input_domain: WildExprDomain,
+    input_metric: M,
+    expr: Expr,
+) -> Fallible<Transformation<WildExprDomain, ExprDomain, M, M>>
+where
+    M::InnerMetric: DatasetMetric,
+    M::Distance: Clone,
+    (WildExprDomain, M): MetricSpace,
+    (ExprDomain, M): MetricSpace,
+    Expr: StableExpr<M, M>,
+{
+    let Expr::Function {
+        input: inputs,
+        function: FunctionExpr::StringExpr(StringFunction::Strptime(to_type, mut strptime_options)),
+        ..
+    } = expr
+    else {
+        return fallible!(MakeTransformation, "expected str.strptime expression");
+    };
+
+    let Ok([input, ambiguous]) = <&[_; 2]>::try_from(inputs.as_slice()) else {
+        return fallible!(
+            MakeTransformation,
+            "str.strptime must have two arguments, found {}",
+            inputs.len()
+        );
+    };
+
+    let t_prior = input.clone().make_stable(input_domain, input_metric)?;
+    let (middle_domain, middle_metric) = t_prior.output_space();
+
+    if strptime_options.format.is_none() {
+        return fallible!(MakeTransformation, "format must be specified; otherwise Polars will attempt to infer the format from the data, resulting in an unstable transformation");
+    }
+
+    if matches!(to_type, DataType::Time) && !strptime_options.exact {
+        return fallible!(
+            MakeTransformation,
+            "non-exact not implemented for Time data type"
+        );
+    }
+
+    // Strict casting makes the transformation unstable: errors tell you things about private data.
+    // Could throw an error if strict, but it is the default, so for ease-of-use it is forced to be non-strict.
+    // It is also ok for overflow to wraparound (but may be undesirable for users).
+    strptime_options.strict = false;
+
+    // never raise on error
+    let ambiguous = lit(match literal_value_of::<String>(ambiguous) {
+        Ok(Some(a)) if ["earliest", "latest"].contains(&a.as_str()) => a,
+        _ => "null".to_string(),
+    });
+
+    let mut output_domain = middle_domain.clone();
+    let series_domain = &mut output_domain.column;
+    let name = series_domain.field.name.as_ref();
+
+    // check input and output types
+    if series_domain.field.dtype != DataType::String {
+        return fallible!(
+            MakeTransformation,
+            "str.strptime input dtype must be String, found {}",
+            series_domain.field.dtype
+        );
+    }
+
+    if !matches!(
+        to_type,
+        DataType::Time
+            | DataType::Datetime(TimeUnit::Microseconds | TimeUnit::Milliseconds, _)
+            | DataType::Date
+    ) {
+        return fallible!(
+            MakeTransformation,
+            "str.strptime output dtype must be Time, micro- or milli-second Datetime or Date, found {}",
```

Comment:
The comment above also mentions nanoseconds. Add to this message?

Reply:
Oh, the absence of nano is probably related to https://github.com/pola-rs/polars/issues/19928. Maybe file an issue in this repo to track that?

Reply:
I left an error condition in the code.